pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -262,6 +262,12 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListeners(env *model.En
 					ForwardClientCertDetails: http_conn.APPEND_FORWARD,
 				},
 			}
+			// See https://github.com/grpc/grpc-web/tree/master/net/grpc/gateway/examples/helloworld#configure-the-proxy
+			if endpoint.ServicePort.Protocol == model.ProtocolGRPCWeb {
+				httpOpts.connectionManager.Http2ProtocolOptions = &core.Http2ProtocolOptions{}
```

Comment:
why?

Reply:
That's needed for an internal protocol upgrade.

Reply:
Doesn't envoy auto upgrade for H2?